### PR TITLE
#5: Price

### LINKED ISSUE #5: Price Vue component
## Description

Build a `ripe-price` component that will simply show the current price (including the currency symbol) as text.

Props:
- ripeSDK
- currency

## Reference

https://github.com/ripe-tech/ripe-white/blob/master/vue/components/overrides/white-manager/white-manager.vue#L124

### PR DESCRIPTION
**Pull request by BeeMargarida**
| - | - |
| --- | --- |
| Issue | #5 |
| Dependencies | -- |
| Decisions | Price component, receives configuration for building a RIPE SDK instance, as well as the possibility to receive one. It updates the price according to the currency given, which is reactive. |
| Animated GIF | price-complex-usd <br>price-complex-eur |

### CODE REVIEW

File: vue/components/organisms/ripe-price/ripe-price.vue [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+            },
+            immediate: true
+        }
+    },
+    computed: {
+        priceText() {
+            return this.formatMoney(this.price, this.currency);
+        }
+    },
+    mounted: async function() {
```
Comment (NFSS10) [inline vue/components/organisms/ripe-price/ripe-price.vue]: `created: async function() {`

File: vue/components/organisms/ripe-price/ripe-price.vue [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        async setupRipe() {
+            if (!this.ripeData) {
+                this.ripeData = new Ripe();
+            }
+
+            await this.configRipe();
+
+            if (!global.ripe) {
+                global.ripe = this.ripeData;
+            }
```
Comment (NFSS10) [inline vue/components/organisms/ripe-price/ripe-price.vue:169]: Are we sure we want this? Wouldn't this have the undesired effect of overriding `global.ripe` when we didn't want that to happen?

File: vue/components/organisms/ripe-price/ripe-price.vue [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+            handler: function(value) {
+                if (value) this.$emit("loading");
+                else this.$emit("loaded");
+            },
+            immediate: true
+        }
+    },
+    computed: {
+        priceText() {
+            return this.formatMoney(this.price, this.currency);
```
Comment (NFSS10) [inline vue/components/organisms/ripe-price/ripe-price.vue:117]: We should probably have `this.currency` be always in uppercase as if it isn't in uppercase, the symbol isn't added. Ex:
`EUR` -> € but `eur` -> eur

File: vue/components/organisms/ripe-price/ripe-price.vue [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+    },
+    watch: {
+        currency: {
+            handler: async function(value) {
+                await this.configRipe();
+            }
+        },
+        price: {
+            handler: function(value) {
+                this.$emit("update:price", this.formatMoney(value, this.currency));
```
Comment (NFSS10) [inline vue/components/organisms/ripe-price/ripe-price.vue]: this.$emit("update:price", this.priceText);

File: vue/components/organisms/ripe-price/ripe-price.vue [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+            if (!global.ripe) {
+                global.ripe = this.ripeData;
+            }
+        },
+        onPriceChange(value) {
+            this.price = value.total.price_final;
+        }
+    },
+    destroyed: async function() {
+        if (this.priceBind) await this.ripeData.unbind("price", this.priceBind);
```
Comment (NFSS10) [inline vue/components/organisms/ripe-price/ripe-price.vue]: unbind isn't an `async` method, you can remove the `await`

**Review by NFSS10 — CHANGES_REQUESTED** [review]
BTW, I think we can fix your problem with the tests

**Review by NFSS10 — APPROVED** [review]

**Review by NFSS10 — APPROVED** [review]

**Review by joamag — APPROVED** [review]

**Pull request merged**